src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -678,12 +678,21 @@ private static String getMessageBundle(String className) {
         final String messageBundle;
         final String messages = "messages";
         final int endIndex = className.lastIndexOf('.');
+        final Map<String, String> messageBundleMappings = new HashMap<>();
```

Comment:
Not sure what pieces Map<> and HashMap<> do and mean. In addition, I'm a bit confused what "messageBundleMappings" might imply as a variable. Could you please explain or give a reference to these keywords so I can learn more about them?

Reply:
https://docs.oracle.com/javase/tutorial/java/generics/index.html
please read several articles/pages , keep press "Next".
let me know if you still need some help to explain something.

Reply:
> Not sure what pieces Map<> and HashMap<>

https://java-programming.mooc.fi/part-8/2-hash-map 

HashMap is a part of the Java collections it is basically the same as HashTable data structure. It stores data in key-value pairs and `Map<String, String>` means that the key and the values are both Strings

btw thanks for your YouTube videos it will be a great help for new contributors. I was struggling for hours with these  basic things :) .

Reply:
> > Not sure what pieces Map<> and HashMap<>
> 
> https://java-programming.mooc.fi/part-8/2-hash-map
> 
> HashMap is a part of the Java collections it is basically the same as HashTable data structure. It stores data in key-value pairs and `Map<String, String>` means that the key and the values are both Strings
> 
> btw thanks for your YouTube videos it will be a great help for new contributors. I was struggling for hours with these basic things :) .

Glad it helps :)